src/amo/reducers/addonsByAuthors.js

```diff
@@ -0,0 +1,126 @@
+/* @flow */
+import type { AddonType } from 'core/types/addons';
+
+
+type State = {
+  byAddonSlug: { [string]: AddonType },
+};
+
+export const initialState: State = {
+  byAddonSlug: {},
+};
+
+export const OTHER_ADDONS_BY_AUTHORS_PAGE_SIZE = 6;
+
+// See for further information about this notation:
+// https://github.com/mozilla/addons-frontend/pull/3027#discussion_r137661289
+export const FETCH_OTHER_ADDONS_BY_AUTHORS: 'FETCH_OTHER_ADDONS_BY_AUTHORS'
+  = 'FETCH_OTHER_ADDONS_BY_AUTHORS';
+export const LOAD_OTHER_ADDONS_BY_AUTHORS: 'LOAD_OTHER_ADDONS_BY_AUTHORS'
+  = 'LOAD_OTHER_ADDONS_BY_AUTHORS';
+
+type FetchOtherAddonsByAuthorsParams = {|
+  errorHandlerId: string,
+  addonType: string,
+  authors: Array<string>,
+  slug: string,
+|};
+
+type FetchOtherAddonsByAuthorsAction = {|
+  type: typeof FETCH_OTHER_ADDONS_BY_AUTHORS,
+  payload: FetchOtherAddonsByAuthorsParams,
+|};
+
+export const fetchOtherAddonsByAuthors = (
+  { errorHandlerId, slug, addonType, authors }: FetchOtherAddonsByAuthorsParams
+): FetchOtherAddonsByAuthorsAction => {
+  if (!errorHandlerId) {
+    throw new Error('An errorHandlerId is required');
+  }
+
+  if (!slug) {
+    throw new Error('An add-on slug is required.');
+  }
+
+  if (!addonType) {
+    throw new Error('An add-on type is required.');
+  }
+
+  if (!authors) {
+    throw new Error('Authors are required.');
+  }
+
+  if (!Array.isArray(authors)) {
+    throw new Error('The authors parameter must be an array.');
+  }
+
+  return {
+    type: FETCH_OTHER_ADDONS_BY_AUTHORS,
+    payload: {
+      errorHandlerId,
+      slug,
+      addonType,
+      authors,
+    },
+  };
+};
+
+type LoadOtherAddonsByAuthorsParams = {|
+  slug: string,
+  addons: Array<AddonType>,
+|};
+
+type LoadOtherAddonsByAuthorsAction = {|
+  type: typeof LOAD_OTHER_ADDONS_BY_AUTHORS,
+  payload: LoadOtherAddonsByAuthorsParams,
+|};
+
+export const loadOtherAddonsByAuthors = (
+  { slug, addons }: LoadOtherAddonsByAuthorsParams
+): LoadOtherAddonsByAuthorsAction => {
+  if (!slug) {
+    throw new Error('An add-on slug is required.');
+  }
+
+  if (!addons) {
+    throw new Error('A set of add-ons is required.');
+  }
+
+  return {
+    type: LOAD_OTHER_ADDONS_BY_AUTHORS,
+    payload: { slug, addons },
+  };
+};
+
+type Action =
+  // This allows to set a default value to the `action` argument in the reducer.
+  | FetchOtherAddonsByAuthorsAction
+  | LoadOtherAddonsByAuthorsAction;
+
+const reducer = (
+  state: State = initialState,
+  action: Action
+): State => {
+  const payload: Object = action.payload || {};
+
+  switch (action.type) {
+    case FETCH_OTHER_ADDONS_BY_AUTHORS:
+      return state;
```

Comment:
I think this should reset the loaded add-ons in case there are any:

```js
return {
  ...state,
  byAddonSlug: {
    ...state.byAddonSlug,
    [payload.slug]: undefined,
  },
};
```